packages/sitecore-jss-nextjs/src/services/graphql-sitemap-service.ts

```diff
@@ -1,258 +1,259 @@
-import chalk from 'chalk';
 import { GraphQLRequestClient } from '@sitecore-jss/sitecore-jss';
-import { StaticPath } from '../sharedTypes/sitemap';
 
-export type GraphQLSitemapServiceConfig = {
+export type StaticPath = {
+  params: {
+    path: string[];
+  };
+  locale?: string;
+};
+
+// TODO: There is some code duplication between here and graphql-dictionary-service;
+// this can be improved by providing a generic way to call the "search" query (Anastasiya, March 2021)
+
+const DEFAULTS = Object.freeze({
+  pageSize: 10,
+});
+
+const query = `
+query SitePageQuery(
+  $rootItemId: String!,
+  $language: String!,
+  $pageSize: Int = 10,
+  $after: String
+) {
+  search(
+    where: {
+      AND:[
+        { name: "_path",      value: $rootItemId },
+        { name: "_language",  value: $language   },
+        { name: "_hasLayout", value: "true"      }
+      ]
+    }
+    first: $pageSize
+    after: $after
+  ) {
+    total
+    pageInfo {
+      endCursor
+      hasNext
+    }
+    results {
+      url {
+        path
+      }
+    }
+  }
+}`;
+
+/**
+ * Configuration options for @see GraphQLDictionaryService instances
+ */
+export interface GraphQLSitemapServiceConfig {
   /**
    * Your Graphql endpoint
    */
   endpoint: string;
-};
 
-type SearchResult = {
+  /**
+   * How many dictionary items to fetch in each GraphQL call. This is needed for pagination.
+   * @default 10
+   */
+  pageSize?: number;
+}
+
+/**
+ * A reply from the GraphQL endpoint for the 'SitePageQuery' query
+ */
+type SitePageQueryResult = {
   search: {
-    results: { url: { path: string } }[];
+    pageInfo: {
+      endCursor: string;
```

Comment:
@anastasiya29 Can you add jsdoc description for this property and may be `hasNext`?

Reply:
Done